sdk/js/src/sui/publish.ts

```diff
@@ -0,0 +1,257 @@
+import {
+  fromB64,
+  isValidSuiAddress,
+  normalizeSuiObjectId,
+  TransactionBlock,
+} from "@mysten/sui.js";
+import { execSync } from "child_process";
+import fs from "fs";
+import path from "path";
+import { Network } from "../utils";
+import { MoveToml } from "./MoveToml";
+import { GithubTreeResponse, SuiBuildOutput } from "./types";
+
+export const publishCoin = async (
+  network: Network,
+  coreBridgeAddress: string,
+  tokenBridgeAddress: string,
+  vaa: string,
+  signerAddress: string
+): Promise<TransactionBlock> => {
+  await cloneDependencies();
+  setupMainToml(
+    `${__dirname}/dependencies/wormhole`,
+    network,
+    coreBridgeAddress
+  );
+  setupMainToml(
+    `${__dirname}/dependencies/token_bridge`,
+    network,
+    tokenBridgeAddress
+  );
+  setupCoin(coreBridgeAddress, tokenBridgeAddress, vaa);
+  const buildOutput = buildPackage(`${__dirname}/wrapped_coin`);
+  const tx = publishPackage(buildOutput, signerAddress);
+  cleanupTempToml(`${__dirname}/dependencies/wormhole`);
+  cleanupTempToml(`${__dirname}/dependencies/token_bridge`);
+  return tx;
+};
+
+const buildPackage = (packagePath: string): SuiBuildOutput => {
+  return JSON.parse(
+    execSync(
+      `sui move build --dump-bytecode-as-base64 --path ${packagePath} 2> /dev/null`,
+      {
+        encoding: "utf-8",
+      }
+    )
+  );
+};
+
+const cleanupTempToml = (packagePath: string): void => {
+  const defaultTomlPath = getDefaultTomlPath(packagePath);
+  const tempTomlPath = getTempTomlPath(packagePath);
+  if (fs.existsSync(tempTomlPath)) {
+    fs.renameSync(tempTomlPath, defaultTomlPath);
+  }
+};
+
+const cloneDependencies = async (): Promise<void> => {
+  const { tree, sha: latestSha } = await fetchWormholeTree(
+    "sui/integration_v2"
```

Comment:
Can we make this a const at the top of the file so it's clear that we'll need to change this?

It might be best if `fetchWormholeTree` took an optional argument for the ref, which will default to `main`. And for the time being, we are explicitly setting it to a const value (e.g. `TEMPORARY_SUI_BRANCH`) using that const value here.

Either that or a TODO comment would be helpful.

Reply:
Hoisted the const and added TODO comment, didn't add the `main` default param since we will require a code change to remove the usage of that const anyway.